Allow setting processing workflow type

jraddaoui opened the pull request:
- Allow setting processing workflow type from watcher configuration.
- Add watcher configuration defaults and validation.
- Remove AutoApproveAIP option from workflow request and pass type.
- Consider workflow type in processing workflow.
- Remove `unspecified` value and update workflow type enum to a string.
- Update DB schema, generate Ent code and add migration.
- Regenerate API and dashboard client code.

Refs #1218.

jraddaoui committed:
Allow setting processing workflow type

- Allow setting processing workflow type from watcher configuration.
- Add watcher configuration defaults and validation.
- Remove AutoApproveAIP option from workflow request and pass type.
- Consider workflow type in processing workflow.
- Remove `unspecified` value and update workflow type enum to a string.
- Update DB schema, generate Ent code and add migration.
- Regenerate API and dashboard client code.

jraddaoui commented on dashboard/src/main.ts:
```diff
@@ -50,9 +50,9 @@ app.config.globalProperties.$filters = {
+      case api.EnduroIngestSipWorkflowTypeEnum.Aip:
         return "Create AIP";
-      case api.EnduroIngestSipWorkflowTypeEnum.CreateAndReviewAip:
+      case api.EnduroIngestSipWorkflowTypeEnum.AndReviewAip:
```
Don't ask me why ;)

I tried re-generating the client code more than once, but those are the types it chooses.

djjuhasz replied:
I figured this out: the OpenAPI generator defaults to removing identical prefixes from enum names ("removeEnumValuePrefix").  I created a new branch with the fix so I didn't mess with your PR: https://github.com/artefactual-sdps/enduro/compare/dev/issue-1218-fix-dashboard-workflow-types

Feel free to merge the fix if that works for you.

jraddaoui replied:
Niiice, thanks for the detective work!

jraddaoui commented on dashboard/src/pages/ingest/sips/[id]/index.vue:
```diff
@@ -23,12 +22,7 @@ const toggleHelp = () => {
-      (w) =>
-        w.type === api.EnduroIngestSipWorkflowTypeEnum.CreateAip ||
-        w.type === api.EnduroIngestSipWorkflowTypeEnum.CreateAndReviewAip,
-    )[0],
+  () => sipStore.currentWorkflows?.workflows?.[0],
 );
```
Kind of unnecessary, it just makes the template a bit cleaner.

jraddaoui commented on internal/db/migrations/atlas.sum:
```diff
@@ -9,4 +9,5 @@ h1:4nZSKvUlpsK5RT4Ctn1jUnwvnDQRGaEYafWJxJ7s95w=
 20250418000742_update_sip_statuses.up.sql h1:FUbEzeO8+t0FVImqnOWApQqKzxulq2GY4urMSsJlbc0=
 20250509030046_restore_sip_status_index.up.sql h1:8iKBW+nC9mgEs6PqQJ4Frq1IGJOjedAyhm3dO27yVZw=
-20250509030521_add_sip_uuid_column.up.sql h1:z5zEsQbzL/nz4q7MgBdkiQa+KKB5rPopYrte0lRrst4=
+20250509030521_add_sip_uuid_column.up.sql h1:rtcC2HuAOJZI+/7ZkWrlrTn+wkrZ34BRqLK5bJRo3h8=
```
Looks like the previous hash was not right, I'll think about adding CI to check for missing migrations and things like this.

djjuhasz replied:
I think adding a CI check for a mismatched checksum is a good idea.  Atlas has a [migrate validate command](https://atlasgo.io/cli-reference#atlas-migrate-validate) that we could use.

djjuhasz commented on internal/watcher/config.go:
```diff
@@ -13,6 +17,49 @@ type Config struct {
+	}
+}
+
+func (c Config) Validate() error {
```
Nice! I like the config validation. :+1:

djjuhasz committed:
Fix dashboard ingest workflow type keys

- Turn off OpenAPI generator "removeEnumValuePrefix" setting
- Regenerate dashboard models using the OpenAPI generator

djjuhasz reviewed (approved):
This looks really good @jraddaoui, thanks.

I figured out why the OpenAPI generator was removing the "create" prefix from the ingest workflow type enum keys.  It's not a critical fix, but I do think it makes the dashboard code easier to understand.
